components/calender/index.tsx

```diff
@@ -0,0 +1,143 @@
+  const startXRef = useRef(null);
+
+  const handleMouseDown = (event: any) => {
+    startXRef.current = event.clientX;
+  };
+
+  const handleMouseUp = (event: any) => {
+    if (startXRef.current === null) {
+      return;
+    }
+    const endX = event.clientX;
+    const diffX = endX - startXRef.current;
+
+    if (diffX > 50) {
+      handlePrevMonth();
+    } else if (diffX < -50) {
+      handleNextMonth();
+    }
+
+    startXRef.current = null;
+  };
+
+  const handleTouchStart = (event: any) => {
+    startXRef.current = event.touches[0].clientX;
+  };
+
+  const handleTouchEnd = (event: any) => {
+    if (startXRef.current === null) {
+      return;
+    }
+    const endX = event.changedTouches[0].clientX;
+    const diffX = endX - startXRef.current;
+
+    if (diffX > 50) {
+      handlePrevMonth();
+    } else if (diffX < -50) {
+      handleNextMonth();
+    }
+  };
+
+  const handlePrevMonth = () => {
+    if (currentMonth === 0) {
+      setCurrentMonth(11);
+      setCurrentYear(currentYear - 1);
+    } else {
+      setCurrentMonth(currentMonth - 1);
+    }
+  };
+
+  const handleNextMonth = () => {
+    if (currentMonth === 11) {
+      setCurrentMonth(0);
+      setCurrentYear(currentYear + 1);
+    } else {
+      setCurrentMonth(currentMonth + 1);
+    }
+  };
+
+  const daysInMonth = new Date(currentYear, currentMonth + 1, 0).getDate();
+  const firstDayOfMonth = new Date(currentYear, currentMonth, 1).getDay();
+
+  const handleDateClick = (dayNumber: any) => {
+    console.log(`Clicked on ${currentYear}-${currentMonth + 1}-${dayNumber}`);
+  };
+  return (
+    <div>
+      <button
+        type="button"
+        className="container mx-auto mt-5"
+        ref={calendarRef}
+        onMouseDown={handleMouseDown}
+        onMouseUp={handleMouseUp}
+        onTouchStart={handleTouchStart}
+        onTouchEnd={handleTouchEnd}
+      >
+        <div className="text-center">
+          <h1 className="text-3xl font-semibold mb-2">
+            {currentYear}년 {currentMonth + 1}월
+          </h1>
+          <p>한국 시간: {koreaDateTime}</p>
+        </div>
+        <div className="mt-5 grid grid-cols-7 gap-4 m-12-500">
+          {daysOfWeek.map(day => (
+            <div key={day} className="text-center p-2">
+              {day}
+            </div>
+          ))}
+          {Array.from({ length: firstDayOfMonth }, (_, index) => (
+            <div key={`empty-${index}`} className="p-2" />
+          ))}
+          {Array.from({ length: daysInMonth }, (_, index) => {
+            const dayNumber = index + 1;
+            return (
+              <div
+                key={`day-${index}`}
+                className={`flex-center p-1 ${
+                  dayNumber > 0 && dayNumber <= daysInMonth
+                    ? "bg-gray-200 cursor-pointer rounded-lg ring-1 ring-gray-300 w-8 h-8"
+                    : ""
+                }`}
+              >
+                {dayNumber > 0 && dayNumber <= daysInMonth && (
```

Comment:
```suggestion
                {dayNumber > 0 && dayNumber <= daysInMonth && (
```
해당 로직 3군데에서 반복적으로 사용중인 boolean 인데 의도를 알 수 있게 따로 빼서 사용하는게 좋아보입니👍